src/haxelib/api/Vcs.hx

```diff
@@ -356,14 +358,39 @@ class Git extends Vcs {
 		Sys.setCwd(libPath);
 
 		final ret: VcsData = {
+			// `git config --get remote.remoteName.url` will get the original url of the remote
+			// for usecases where someone has something like `url."git@github.com:".insteadOf 'https://github.com/'` set
+
 			// could the remote's name not be "origin"?
-			url: run(["remote", "get-url", "origin"], true).out.trim(),
+			url: run(["config", "--get", "remote.origin.url"], true).out.trim(),
 			ref: run(["rev-parse", "HEAD"], true).out.trim(),
 		};
 
 		Sys.setCwd(oldCwd);
 		return ret;
 	}
+
+	public function parseUrl(url:String):{protocol:String, domain:String, user:String, repo:String, cleanUrl:String}
+	{
+		var regex = new EReg("^(https:\\/\\/|git@)([^\\s/:]+)(?::|\\/)([^\\s/]+)\\/([^\\s/]+?)(?:\\.git)?$", "");
```

Comment:
I would prefer to avoid complicated parsing like this, especially because this will only work with github repository urls. I think it should be good enough to cover cases where the url matches identically.

Reply:
that sounds good, I suppose I was aiming to get it a bit cross compatible with ssh vs. git. If someone wants to download the git@ ssh version of a git repo, but their remote is already at the https version of a git repo, we might want to just update the remote rather than redownload. However just using identical urls sounds good